### PR TITLE
S2U-39 5.2.3.1 Wiki: Add group support to Wiki pages - fix

### PR DESCRIPTION
**Pull request by JuanDavid102**
This is linked with https://github.com/sakaiproject/sakai-reference/pull/209

### CODE REVIEW

**Review by mpellicer — CHANGES_REQUESTED** [review]
This requires a conversion script, please link it in sakai-reference

File: rwiki/rwiki-api/api/src/java/uk/ac/cam/caret/sakai/rwiki/service/api/model/RWikiObject.java [hunk 1]
```diff
@@ -337,30 +339,30 @@ public interface RWikiObject extends Comparable
...
-	String getPageGroups();
+	void setPageGroupsAsList(List pageGroups);
 
 	/**
-	 * @return The pageGroups list for this page
+	 * @return The pageGroups List for this page
 	 * @return
 	 */
-	String[] getPageGroupsAsArray();
+	List getPageGroupsAsList();
```
Comment (mpellicer) [inline rwiki/rwiki-api/api/src/java/uk/ac/cam/caret/sakai/rwiki/service/api/model/RWikiObject.java:365]: Can we define the type? List<String> ?

File: rwiki/rwiki-api/api/src/java/uk/ac/cam/caret/sakai/rwiki/service/api/model/RWikiObject.java [hunk 2]
```diff
@@ -337,30 +339,30 @@ public interface RWikiObject extends Comparable
...
 
 	/**
-	 * @return The pageGroups String for this page
-	 * @return
+	 * The pageGroups List for this page
+	 * 
+	 * @param pageGroups
 	 */
-	String getPageGroups();
+	void setPageGroupsAsList(List pageGroups);
```
Comment (mpellicer) [inline rwiki/rwiki-api/api/src/java/uk/ac/cam/caret/sakai/rwiki/service/api/model/RWikiObject.java:359]: Can we define the type?

`List<String>`

File: rwiki/rwiki-model/src/java/uk/ac/cam/caret/sakai/rwiki/model/RWiki.hbm.xml [hunk 3]
```diff
@@ -103,6 +103,9 @@
 		<property name="publicWrite" type="boolean" />
 		<property name="sha1" length="64" />
 		<property name="revision"  type="int"/>
-		<property name="pageGroups" column="page_groups" type="materialized_clob" />
+		<set name="pageGroups" table="rwikipagegroups" cascade="all" lazy="false">
```
Comment (mpellicer) [inline rwiki/rwiki-model/src/java/uk/ac/cam/caret/sakai/rwiki/model/RWiki.hbm.xml:106]: Nice!

Comment (mpellicer) [reply rwiki/rwiki-model/src/java/uk/ac/cam/caret/sakai/rwiki/model/RWiki.hbm.xml:106]: We need a conversion script for this.

File: rwiki/rwiki-api/api/src/java/uk/ac/cam/caret/sakai/rwiki/service/api/model/RWikiObject.java (hunk 1 above)
Comment (JuanDavid102) [reply rwiki/rwiki-api/api/src/java/uk/ac/cam/caret/sakai/rwiki/service/api/model/RWikiObject.java]: Yes

**Comment by JuanDavid102** [comment]
The requested conversion scripts its here: [](https://github.com/sakaiproject/sakai-reference/pull/209)

**Comment by mpellicer** [comment]
Could you check everything here @bgarciaentornos ? I've almost reviewed it completelly but there are DB changes.

**Review by bgarciaentornos — APPROVED** [review]
I don't find any code issue, LGTM

**Comment by mpellicer** [comment]
Thanks for all the great work on this.

**Pull request merged**